test/test_sparse.py

```diff
@@ -1694,6 +1694,129 @@ def test_log1p(self):
+        self.assertEqual(expected_output, sparse_tensor.negative().to_dense())
+        sparse_tensor_copy = sparse_tensor.clone()
+        self.assertEqual(expected_output, sparse_tensor_copy.negative_().to_dense())
+
+    def test_neg_negative(self):
+
+        if not self.is_uncoalesced:
+            input_coalesced = torch.sparse_coo_tensor(
+                indices=torch.tensor([[0, 1, 2]]),
+                values=torch.tensor([3.0, -4.0, 5.0]),
+                size=[3, ],
+                device=self.device
+            )
+            self._test_neg_negative(input_coalesced)
+
+            # hybrid sparse input
+            input_coalesced = torch.sparse_coo_tensor(
+                indices=torch.tensor([[1, 3], [2, 4]]),
+                values=torch.tensor([[-1.0, 3.0], [-5.0, 7.0]]),
+                size=[4, 5, 2],
+                device=self.device
+            )        
+            self._test_neg_negative(input_coalesced)
+
+        if self.is_uncoalesced:
+            # test uncoalesced input
+            input_uncoalesced = torch.sparse_coo_tensor(
+                indices=torch.tensor([[0], [1], [2], [0], [1], [2]]).transpose(1, 0),
+                values=torch.tensor([2.0, -3.0, -4.0, 1.0, -1.0, 1.5]),
+                size=[3, ],
+                device=self.device
+            )
+            self._test_neg_negative(input_uncoalesced)
+
+            # test on empty sparse tensor
+            input_uncoalesced = torch.sparse_coo_tensor(
+                indices=torch.zeros([2, 0]),
+                values=torch.zeros([0, 5, 5, 5, 5, 5, 5, 0]),
+                size=[0, 0, 5, 5, 5, 5, 5, 5, 0],
+                device=self.device
+            )
+            self._test_neg_negative(input_uncoalesced)
+
+    def _test_asin_arcsin(self, sparse_tensor):
+        dense_tensor = sparse_tensor.to_dense()
+        expected_output = dense_tensor.asin()
+        self.assertEqual(expected_output, torch.asin(sparse_tensor).to_dense())
+        sparse_tensor_out = torch.zeros_like(sparse_tensor)
+        torch.asin(sparse_tensor, out=sparse_tensor_out)
+        self.assertEqual(expected_output, sparse_tensor_out.to_dense())
+        self.assertEqual(expected_output, sparse_tensor.asin().to_dense())
+        self.assertEqual(expected_output, sparse_tensor.coalesce().asin_().to_dense())
+
+        self.assertEqual(expected_output, torch.arcsin(sparse_tensor).to_dense())
+        sparse_tensor_out = torch.zeros_like(sparse_tensor)
+        torch.arcsin(sparse_tensor, out=sparse_tensor_out)
+        self.assertEqual(expected_output, sparse_tensor_out.to_dense())
+        self.assertEqual(expected_output, sparse_tensor.arcsin().to_dense())
+        self.assertEqual(expected_output, sparse_tensor.coalesce().arcsin_().to_dense())
+
+        # test in-place op on uncoalesced input
+        with self.assertRaisesRegex(RuntimeError, "in-place on uncoalesced tensors is not supported"):
```

Comment:
How is it that this knows `sparse_tensor` is uncoalesced? I thought when `self.is_uncoalesced` is False that coalesced sparse tensors were passed to this function? Also: shouldn't there be a test for the inplace version of the op when the tensor is coalesced?

Reply:
Thanks for catching that !